app/src/main/java/org/oppia/app/options/OptionsFragmentPresenter.kt

```diff
@@ -0,0 +1,244 @@
+  private val fragment: Fragment,
+  private val profileManagementController: ProfileManagementController,
+  private val viewModelProvider: ViewModelProvider<OptionControlsViewModel>
+) {
+  private lateinit var binding: OptionsFragmentBinding
+
+  private lateinit var recyclerViewAdapter: RecyclerView.Adapter<*>
+  private var internalProfileId: Int = -1
+  private lateinit var profileId: ProfileId
+  private var storyTextSize = StoryTextSize.SMALL_TEXT_SIZE
+  private var appLanguage = AppLanguage.ENGLISH_APP_LANGUAGE
+  private var audioLanguage = AudioLanguage.NO_AUDIO
+
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+    binding = OptionsFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+    val viewModel = getOptionControlsItemViewModel()
+
+    internalProfileId = activity.intent.getIntExtra(KEY_NAVIGATION_PROFILE_ID, -1)
+    profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()
+    viewModel.setProfileId(profileId)
+
+    val optionsRecyclerViewAdapter = createRecyclerViewAdapter()
+    binding.optionsRecyclerview.apply {
+      adapter = optionsRecyclerViewAdapter
+    }
+    recyclerViewAdapter = optionsRecyclerViewAdapter
+    binding.let {
+      it.lifecycleOwner = fragment
+      it.viewModel = viewModel
+    }
+    return binding.root
+  }
+
+  private fun createRecyclerViewAdapter(): BindableAdapter<OptionsItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<OptionsItemViewModel, ViewType> { viewModel ->
+        when (viewModel) {
+          is OptionsStoryTextViewViewModel -> ViewType.VIEW_TYPE_STORY_TEXT_SIZE
+          is OptionsAppLanguageViewModel -> ViewType.VIEW_TYPE_APP_LANGUAGE
+          is OptionsAudioLanguageViewModel -> ViewType.VIEW_TYPE_AUDIO_LANGUAGE
+          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
+        }
+      }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_STORY_TEXT_SIZE,
+        inflateDataBinding = OptionStoryTextSizeBinding::inflate,
+        setViewModel = this::bindStoryTextSize,
+        transformViewModel = { it as OptionsStoryTextViewViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_APP_LANGUAGE,
+        inflateDataBinding = OptionAppLanguageBinding::inflate,
+        setViewModel = this::bindAppLanguage,
+        transformViewModel = { it as OptionsAppLanguageViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_AUDIO_LANGUAGE,
+        inflateDataBinding = OptionAudioLanguageBinding::inflate,
+        setViewModel = this::bindAudioLanguage,
+        transformViewModel = { it as OptionsAudioLanguageViewModel }
+      )
+      .build()
+  }
+
+  private fun bindStoryTextSize(binding: OptionStoryTextSizeBinding, model: OptionsStoryTextViewViewModel) {
+    binding.viewModel = model
+    binding.storyTextSizeTextView.setOnClickListener {
```

Comment:
Just noticed this major issue:

Actually the click listener should be set on the entire item and not just the text.

You will need to update the test cases too.

Reply:
I have updated listener and testcases related to it. PTAL.